shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -45,18 +100,185 @@ impl RedisConfig {
+                                        let mut cursor = Cursor::new(redis_bytes.as_ref());
+                                        let answer =
+                                            CassandraResult::from_cursor(&mut cursor, Version::V4);
+                                        if let Ok(result) = answer {
+                                            Ok(result)
+                                        } else {
+                                            Err(CacheableState::Err {
+                                                reason: answer.unwrap_err().to_string(),
+                                            })
+                                        }
+                                    }
+                                    RedisFrame::Array(_) => Err(CacheableState::Err {
+                                        reason: "Redis returned an array value".into(),
+                                    }),
+                                    RedisFrame::Null => {
+                                        self.missed_requests.increment(1);
+                                        Err(CacheableState::Skip {
+                                            reason: "No cache results".into(),
+                                        })
+                                    }
+                                }
+                            }
+
+                            _ => Err(CacheableState::Err {
+                                reason: "No Redis frame in Redis response".into(),
+                            }),
+                        }
+                    } else {
+                        Err(CacheableState::Err {
+                            reason: "Redis response was None".into(),
+                        })
+                    }
+                } else {
+                    Ok(CassandraResult::Void)
+                };
+            if let Err(state) = cassandra_result {
+                return Err(state);
+            }
+
+            *cass_request = Message::from_frame(Frame::Cassandra(CassandraFrame {
+                version: Version::V4,
+                operation: CassandraOperation::Result(cassandra_result.ok().unwrap()),
+                stream_id: cass_request.stream_id().unwrap(),
+                tracing_id: None,
+                warnings: vec![],
+            }));
+        }
+        Ok(cassandra_messages)
+    }
+
+    /// Reads the data from the cache for all the messages or None.
+    /// on success the values in the messages_cass_request will be modified to be Cassandra response messages.
+    /// return is the Cassandra response messages or an error containing CacheableState::Skip or CacheableState::Err.
+    async fn read_from_cache(
+        &mut self,
+        mut cassandra_messages: Messages,
+    ) -> Result<Messages, CacheableState> {
+        // This function is a little hard to follow, so here's an overview.
```

Comment:
This entire comment block is no longer true with your rewrite, please just delete it.